tests/test_model.py

```diff
@@ -0,0 +1,636 @@
+        "display_name": "Test Car",
+        "odometer": 12345.67,
+        "car_version": "2023.10.1"
+    }
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    # Update with API request data
+    api_data = {
+        "tesla_request_counter": {
+            "wakes": 1,
+            "commands": 12
+        }
+    }
+    
+    vehicle.update_non_empty(api_data)
+    
+    assert vehicle.get('wakes', None) == 1
+    assert vehicle.get('commands', None) == 12
+    
+def test_TeslaFiVehicle_id():
+    """Test TeslaFiVehicle class id property"""
+    vehicle_data = {
+        "id": "123.324 324",
+        "vin": "5YJSA1CG3DFP14555",
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.id == "123.324 324"
+
+def test_TeslaFiVehicle_id_use_vin():
+    """Test TeslaFiVehicle class id property"""
+    vehicle_data = {
+        "vin": "5YJSA1CG3DFP14555",
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.id == "5YJSA1CG3DFP14555"
+
+def test_TeslaFiVehicle_vehicle_id():
+    """Test TeslaFiVehicle class id property"""
+    vehicle_data = {
+        "vehicle_id": "FOO_VEHICLE_ID",
+        "vin": "5YJSA1CG3DFP14555",
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.vehicle_id == "FOO_VEHICLE_ID"
+
+def test_TeslaFiVehicle_vehicle_id_use_vin():
+    """Test TeslaFiVehicle class id property"""
+    vehicle_data = {
+        "vin": "5YJSA1CG3DFP14555"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.vehicle_id == "5YJSA1CG3DFP14555"
+
+def test_TeslaFiVehicle_firmware_version():
+    """Test TeslaFiVehicle class id property"""
+    vehicle_data = {
+        "car_version": "2025.2.9"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.firmware_version == "2025.2.9"
+
+def test_TeslaFiVehicle_firmware_version_none():
+    """Test TeslaFiVehicle class firmware_version property"""
+    vehicle_data = {
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.firmware_version is None
+
+def test_TeslaFiVehicle_name():
+    """Test TeslaFiVehicle class name property"""
+    vehicle_data = {
+        "display_name": "Model 3"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.name == "Model 3"
+
+def test_TeslaFiVehicle_name_none():
+    """Test TeslaFiVehicle class name property"""
+    vehicle_data = {
+        "vin": "5YJSA1CG3DFP14555",
+        "display_name": None
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.name == "2013 modelS 4555"
+    
+def test_TeslaFiVehicle_last_remote_update():
+    """Test TeslaFiVehicle class last_remote_update property"""
+    vehicle_data = {
+        "Date": "2023-10-01 12:00:00"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.last_remote_update == datetime.strptime("2023-10-01 12:00:00", TESLAFI_DATE_FORMAT)
```

Comment:
nit: we should be able to use a direct `datetime` comparison here, instead of re-parsing the same string. E.g., if there's a bug in the `TESLAFI_DATE_FORMAT`, this test won't catch that, because they will both have been parsed using the buggy string format.

```suggestion
    assert vehicle.last_remote_update == datetime(2023, 10, 1, 12, 0, 0)
```